modules/server/src/mapping/mappingToFields.js

```diff
@@ -29,7 +29,7 @@ const mappingToFields = ({ enableDocumentHits, type, parent }) => {
 				type.customFields,
 			],
 			createStateTypeDefs: 'createState' in type ? type.createState : true,
-			fieldsToExclude,
+			showRecords,
```

Comment:
given the meaningful ternaries within `createStateTypeDefs` relying on negatives, it may be good to keep the same original name here (`enableDocumentHits`) to facilitate keeping the larger mental context. i.e. changing the name to this synonym doesn't add clarity while it does add complexity, now having to keep in mind "this means that".

...or better yet, make this `hideHits` if you seek to add clarity through simplicity, with a bonus of "not asking negatives" within that other scope.